src/ast/nodes/ExportDefaultDeclaration.ts

```diff
@@ -105,7 +105,7 @@ export default class ExportDefaultDeclaration extends NodeBase {
 			`${this.module.graph.varOrConst} ${this.variable.getName()} = ${systemBinding}`
 		);
 		if (systemBinding) {
-			code.prependRight(this.end - 1, ')');
+			code.appendRight(code.original[this.end - 1] === '}' ? this.end : this.end - 1, ')');
```

Comment:
I do not think this is the proper fix. If in the `entrypoint-facade` test you change the default export to `export default 42` without a semicolon, the system output becomes `exports('default$1', 4)2`. Instead I would suggest to check for the presence of a semicolon as the last character.

Reply:
Nice, that works, and makes me feel a lot better about the fix as well understanding the issue!